src/facet/selection/_selection.py

```diff
@@ -246,6 +532,7 @@ class LearnerRanker(
     algorithm and optimize hyper-parameters.
     """
 
+    @deprecated(message=f"use class {LearnerRanker2.__name__} instead")
```

Comment:
@j-ittner Do we want to keep old one for 2.0? If yes then in which version do you plan to remove old `LearnerRanker` completely? (this information could be also put in message, like: "`LearnerRanker` is expected be superseded by `LearnerRanker2` in 2.x.x. Please migrate to `LearnerRanker2` instead...")

Reply:
No, we should remove the old LearnerRanker before merging this PR, and replace it with the new one. Let's also think about what to call the new one; we may want to change the name.

I kept the old one in for now so that the unit tests keep running; we may want to port some or all of the current unit tests over to the new LearnerRanker, to then be able to safely remove the old one.

Would you have the time to update the old unit tests?

Reply:
Sure! I see that there are a few tests left that use old `LearnerRanker` - I can take care of that. I will prepare a PR to this branch for that.

Reply:
After merging #318 I've got all tests enabled for the new version.